src/http_tracker_connection.cpp

```diff
@@ -168,10 +168,10 @@ namespace libtorrent
 #if TORRENT_USE_I2P
 			if (i2p && tracker_req().i2pconn)
 			{
-				url += "&ip=";
-				url += escape_string(tracker_req().i2pconn->local_endpoint().c_str()
-					, tracker_req().i2pconn->local_endpoint().size());
-				url += ".i2p";
+				if (tracker_req().i2pconn->local_endpoint().empty())
+					fail_impl(error_code(errors::no_i2p_endpoint), -1, "Waiting for i2p acceptor from SAM bridge", 5);
+				else
+					url += "&ip=" + tracker_req ().i2pconn->local_endpoint () + ".i2p";
```

Comment:
No, it won't be deduced. Local *destination* is not related to anything outgoing. Another acceptor would have another base64 *destination*. There is no way to tell destination other than explicitly communicate it.